### DISCUSSION TITLE
[Idea] Be able to force an old product to show up as a new product

### DISCUSSION THREAD

**Discussion by alfonsolr09** [PrestaShop/PrestaShop#32273] [Ideas]
#### Is your feature request related to a problem?

no

#### Describe the solution you'd like

be able to modify the creation date of a product so that it appears in new products.

When I no longer want to sell a product, what I do is overwrite that with a new one so that I don't have to delete them. but when doing this, prestashop does not put it in the new products section

#### Expected solution (by @MatShir)

Create a new date field: Published Date on the product page.
Once the product is created, the first publication date is set to the product creation date.
`ps_newproducts` module should base the date to the new field

At the upgrade all the creation date of all the product as copied to the Published Date

#### Expected solution (by @jolelievre )

This feature should be handled by the `ps_newproducts` module since it's the one that handles the new product list, it will avoid modifying the core code which doesn't have this responsibility.

Also since we're not sure the feature is interesting for everyone it should be configurable, so the first thing to do is to add a switch to configure if the feature is enabled or not in the module configuration form, something similar to this:

<img width="1569" alt="Capture d’écran 2022-07-21 à 16 45 22" src="https://user-images.githubusercontent.com/13801017/180243161-e11f39ed-ec48-4566-9a38-7ad29697930c.png">

When this configuration is enable, and only when it's enabled, the module can add a DatePicker field into the product form. The scope of this feature is for PrestaShop 8.1 minimum and on the new product page only because it offers more potential to modify the product form content. Thus we can add this date field exactly at the place we want.

I'll leave @PrestaShop/product-team and @PrestaShop/ux-team provide a bit more details about the wording, the place of this extra field and so on.

#### Additional context

something like this:
https://imgur.com/a/zhu7g0y

[a comment by an automated account is not shown]

**Comment by 01park** [comment]
You can change date in the data base and if you don't know how, then they are modules that can change it. Or, just create a duplicate product.

**Comment by alfonsolr09** [comment]
I know, until now I have been modifying it in the 2 tables of the database.  but i would like to be able to modify it from the backoffice, how can i implement the code to do it?

**Comment by 01park** [comment]
I don't know the answer to that, but like I said. You can buy modules that do that. They are mostly not expensive. Will save you a lot of time.

**Comment by alfonsolr09** [comment]
ok, but i would like to see if the idea is feasible to be able to integrate it into the backoffice or at least be able to do it

**Comment by PierreRambaud** [comment]
Hi @alfonsolr09 and thanks for your feedback.

Ping @PrestaShop/prestashop-product-team what do you think?

**Comment by LouiseBonnard** [comment]
Hi @alfonsolr09, thanks for your suggestion! I don't think that it should be a core feature since users can do it in their database or use modules to handle this.

**Comment by alfonsolr09** [comment]
all the data can be modified from the database, so why the name and description of the product if it is in the backoffice?

by that rule of three we eliminate the back office and modify everything from the database


Can you at least indicate how I can integrate the creation date in the backoffice? (i want to try to do it)

**Comment by marionf** [comment]
> When I no longer want to sell a product, what I do is overwrite that with a new one so that I don't have to delete them. but when doing this, prestashop does not put it in the new products section

Why do you do that ? Why not deleting the product that you don't want anymore and creating a new one or duplicating the deleted product ?

**Comment by alfonsolr09** [comment]
because I don't like to jump in the id

**Comment by marsaldev** [comment]
> all the data can be modified from the database, so why the name and description of the product if it is in the backoffice?
> 
> by that rule of three we eliminate the back office and modify everything from the database
> 
> Can you at least indicate how I can integrate the creation date in the backoffice? (i want to try to do it)

Hi @alfonsolr09,
there is already a module that does this stuff: https://addons.prestashop.com/it/product-tab/46213-product-date.html

btw: create your own module that will do that it's quite simple 😊

**Comment by alfonsolr09** [comment]
yes, but why not integrate it into prestashop if it is so easy?

**Comment by alfonsolr09** [comment]
how can I create my module? , the prestashop files are very messy

**Comment by PierreRambaud** [comment]
@marionf I agree with @alfonsolr09 even if the data can be modified by an sql query it's not a proper solution. Maybe add a field to define if the product should be in the "new products" listing.

Example of a scenario we used when I was working for a website selling luxuries goods.
- 1 week: Create X products (bags) but not activated
- 2 week: Create Y products (scarfs) but not activated
- 3 week: Activate scarfs that appear as new in the listing
- 4 week: Deactivate scarfs and activate bags that appear as new in the listing

If you prepare a new category, it can take time to do it manually. 
You maybe want to activate and add them in the "New products" listing at the same time.

**Comment by alfonsolr09** [comment]
I think it would be as simple as making a box to modify the creation date
or even a better idea would be to use the product update date (since I have seen that the update date is also saved in the database, why not use it?

**Comment by LouiseBonnard** [comment]
Or maybe a product should be considered as new based on its activation date rather than on its date of creation?

**Comment by alfonsolr09** [comment]
it may be a good idea, or there may even be a box that if you check it will be put on new products

**Comment by Klemart3D** [comment]
Product "Creation date" or "Activation date" are semantically not a "New from date".
Best solution would be to do something like Magento, a dedicated range of dates for "New product status". And to keep the PS global product parameter "new_days_number", it can be added a checkbox "Keep global configuration, currently defined to: {20} days" that disabling the second date field if it checked (that would be default).

![image](https://user-images.githubusercontent.com/7419525/80624917-af373580-8a4c-11ea-8767-b3dcf0df4294.png)

Currently, I'm using a [MyPresta.eu module](https://mypresta.eu/modules/administration-tools/product-date-change.html) to do it.

**Comment by alfonsolr09** [comment]
i think its a good idea

**Comment by alfonsolr09** [comment]
> La "Fecha de creación" o la "Fecha de activación" del producto no son semánticamente una "Fecha de inicio nueva".
> La mejor solución sería hacer algo como Magento, un rango dedicado de fechas para el "Nuevo estado del producto". Y para mantener el parámetro de producto global PS "new_days_number", se puede agregar una casilla de verificación "Mantener configuración global, actualmente definida a: {20} días" que deshabilita el segundo campo de fecha si está marcado (eso sería predeterminado).
> 
> ![imagen](https://user-images.githubusercontent.com/7419525/80624917-af373580-8a4c-11ea-8767-b3dcf0df4294.png)
> 
> Actualmente, estoy usando un [módulo MyPresta.eu](https://mypresta.eu/modules/administration-tools/product-date-change.html) para hacerlo.

I still think that idea is great, the image is worth a thousand words

**Comment by Hlavtox** [comment]
@alfonsolr09 If you really want to do this, just run a simple query:

```
UPDATE ps_product SET date_upd = '2021-05-27 00:00:00', date_add = '2021-05-27 00:00:00' WHERE id_product = 123yourid123;
UPDATE ps_product_shop SET date_upd = '2021-05-27 00:00:00', date_add = '2021-05-27 00:00:00' WHERE id_product = 123yourid123;
```
**But I think you definitely should NOT overwrite products with new ones.**
- By overwriting a product, you loose your sell history. A customer will call you after 1 year and will want to apply his warranty. You open the order and could see malformed product data.
- With every page you make, you make it indexed and possibly linked from other websites. If you overwrite it and change it, users can land into a nonsense page that does not interest them. If they landed where they wanted and you told them "this is no longer sold, choose something different" they won't be confused.
- IDs are only internal thing, you should not use it for communicating, referencing products, marking then in stock etc. That's what's reference is for. I have 6-7 digit ids, and what, nobody cares. 👍

**Comment by Patrickclicweb** [comment]
Bonjour,
Cela peut-être également utile lors de la mise à jour de quantité de pouvoir le remettre comme "nouveau" si le produit est resté longtemps indisponible, cela remettrait directement le produit dans le slider "nouveauté" 
Le faire en SQl n'est pas une solution pour tous les utilisateurs 
cette fonction était présente il me semble en 1.5

**Comment by alfonsolr09** [comment]
I have a simple module to do it

**Comment by alfonsolr09** [comment]
If you want it send me a message, its free

**Comment by Patrickclicweb** [comment]
OK fait moi toi un message car je ne suis pas très à l'aise avec Github :(

**Comment by MatShir** [comment]
Module can do it. IMO it is a basic and needed feature that can be included into the core. But it won't be prioritized for the next version but however, PrestaShop is an open-source project, if you have the technical knowledge or find someone that could don't hesitate to propose a pull request ([find more information here](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/)). We will be happy to review your contribution to the PrestaShop core.

**Comment by alfonsolr09** [comment]
> OK fait moi toi un message car je ne suis pas très à l'aise avec Github :(

send me a message pedropicaso04@gmail.com
